packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx

```diff
@@ -30,187 +30,158 @@ const messages = defineMessages({
-    if (hasMore) {
-      const offset = this.state.search !== search ? 0 : additional.offset;
-      this.props.getVocabulary({
-        vocabNameOrURL: this.props.vocabBaseUrl,
-        query: search,
-        start: offset,
-      });
-      this.setState({ search });
-
-      return {
-        options:
-          intersection(previousOptions, this.props.choices).length ===
-          this.props.choices.length
-            ? []
-            : this.props.choices,
-        hasMore: hasMore,
-        additional: {
-          offset: offset === additional.offset ? offset + 25 : offset,
-        },
-      };
-    }
-    return null;
-  };
-
-  /* Customized to pass object instead of plain string value */
-  handleChange = (selectedOption) => {
-    this.setState({ selectedOption });
-    this.props.onChange(this.props.id, {
-      value: selectedOption.value,
-      title: selectedOption.label,
-    });
-  };
-
-  /**
-   * Render method.
-   * @method render
-   * @returns {string} Markup for the component.
-   */
-  render() {
-    const {
-      id,
-      // choices,
-      value,
-      onChange,
-      placeholder,
-      querystring,
-      filterOptions = identity,
-    } = this.props;
-    const isDisabled = false;
-    const { indexes = [] } = querystring;
-
-    const Select = this.props.reactSelect.default;
-
-    return (
-      <FormFieldWrapper {...this.props}>
-        <Select
-          id={`field-${id}`}
-          name={id}
-          placeholder={
-            placeholder ?? this.props.intl.formatMessage(messages.select)
-          }
-          isDisabled={isDisabled}
-          className="react-select-container"
-          classNamePrefix="react-select"
-          options={map(
-            toPairs(
-              groupBy(toPairs(filterOptions(indexes)), (item) => item[1].group),
+  }, [choices, vocabBaseUrl, getVocabulary]);
+
+  // const loadOptions = (search, previousOptions, additional) => {
+  //   let hasMore = props.itemsTotal > previousOptions.length;
+  //   if (hasMore) {
+  //     const offset = selectedOption.search !== search ? 0 : additional.offset;
+  //     props.getVocabulary({
+  //       vocabNameOrURL: props.vocabBaseUrl,
+  //       query: search,
+  //       start: offset,
+  //     });
+  //     setSelectedOption({ search });
+
+  //     return {
+  //       options:
+  //         intersection(previousOptions, props.choices).length ===
+  //           props.choices.length
+  //           ? []
+  //           : props.choices,
+  //       hasMore: hasMore,
+  //       additional: {
+  //         offset: offset === additional.offset ? offset + 25 : offset,
+  //       },
+  //     };
+  //   }
+  //   return null;
+  // };
+
+  // const handleChange = (selectedOption) => {
+  //   setSelectedOption(selectedOption);
+  //   props.onChange(props.id, {
+  //     value: selectedOption.value,
+  //     title: selectedOption.label,
+  //   });
+  // };
```

Comment:
Commenting code as you have done twice above is a sloppy development practice and makes reviewing `git diff` difficult. Delete the commented code.

Reply:
ok sir , I will delete the commented code

Reply:
Deleted the commented code